### ISSUE TITLE
GVK search uses potentially unavailable entry types

### ISSUE THREAD

**Issue by stefan-kolb** [JabRef/jabref#447]
Search term examples: "Lothar Papula"

Retrieves entries that get categorized as `Online`. When working with Bibtex mode this entry type is not available inside the `ALL_TYPES` collection.

Output of Error Console:
```
Exception in thread "JabRef CachedThreadPool" java.lang.NullPointerException: Every BibtexEntry must have a type.
at java.util.Objects.requireNonNull(Unknown Source)
at net.sf.jabref.model.entry.BibtexEntry.setType(BibtexEntry.java:125)
at net.sf.jabref.model.entry.BibtexEntry.(BibtexEntry.java:76)
at net.sf.jabref.importer.fetcher.GVKParser.parseEntry(GVKParser.java:367)
at net.sf.jabref.importer.fetcher.GVKParser.parseEntries(GVKParser.java:63)
at net.sf.jabref.importer.fetcher.GVKParser.parseEntries(GVKParser.java:40)
at net.sf.jabref.importer.fetcher.GVKFetcher.fetchGVK(GVKFetcher.java:170)
at net.sf.jabref.importer.fetcher.GVKFetcher.processQuery(GVKFetcher.java:111)
at net.sf.jabref.importer.fetcher.GeneralFetcher$4.run(GeneralFetcher.java:288)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
```

**Comment by stefan-kolb** [comment]
Currenty `EntryType getType()` does only include valid types for the current mode. This issue is also a general problem for all fetchers that must be defined. Will we only fetch to BibTeX entry types or will we     decide based on the current mode?

**Comment by koppor** [comment]
The GVK fetcher is optimizied for BibLaTeX. I see following options:

A. Inform the user if using BibTeX mode and refuse to fetch
B. Fallback to "Misc" at `getType()` when being asked for an unknown type
C. Mixture? - Warn the user and fallback.

Maybe each fetcher should be tagged what modes it supports?

**Comment by stefan-kolb** [comment]
This is a general problem in our current architecture that's why we should discuss what to do.

**Comment by teertinker** [comment]
I like the fallback option. In older jabref versions, gvk entries were often classified as misc. I didnt have any problems whith updating the proper type. Its a lot better than not being able to fetch at all.

**Comment by koppor** [comment]
For the record, a search for "667844562" returns such an entry.

**Comment by koppor** [comment]
I don't recall, why we closed this issue. Is it fixed or do we have to continue working on it? Refs #1069

**Comment by stefan-kolb** [comment]
https://github.com/JabRef/jabref/commit/0a162b7fc3d048b51508e13e3f99fadc39a5f1a9 
Temporary fix for #447 Use misc instead of online entry type
